src/main/java/cms/gov/madie/measure/resources/OrganizationController.java

```diff
@@ -33,13 +37,22 @@ public ResponseEntity<List<Organization>> getAllOrganizations() {
     return ResponseEntity.status(HttpStatus.OK).body(organizationList);
   }
 
-  //  @PostMapping("/organizations")
-  //  public ResponseEntity<List<Organization>> addOrganizations(
-  //      @RequestBody List<Organization> organizations, Principal principal) {
-  //    final String userName = principal.getName();
-  //    log.info("User [{}] is attempting to add new organizations", organizations);
-  //    List<Organization> organizationList = organizationRepository.saveAll(organizations);
-  //    log.info("User [{}] successfully added new organizations", userName);
-  //    return ResponseEntity.status(HttpStatus.CREATED).body(organizationList);
-  //  }
+  @PostMapping("/organizations")
+  @PreAuthorize("#request.getHeader('api-key') == #apiKey")
+  public ResponseEntity<List<Organization>> addOrganizations(
+      HttpServletRequest request,
+      @RequestBody List<Organization> organizations,
+      @Value("${organizations-api-key}") String apiKey) {
+    final String userName = request.getHeader(HARP_ID_HEADER);
+    log.info("User {} is attempting to add new organizations {}", userName, organizations);
+    try {
+      List<Organization> savedOrganizations = organizationRepository.saveAll(organizations);
+      log.info("User {} successfully added new organizations", userName);
+      return ResponseEntity.status(HttpStatus.CREATED).body(savedOrganizations);
+    } catch (org.springframework.dao.DuplicateKeyException duplicateKeyException) {
+      log.error("One of the organizations is already available with same OID");
+      throw new DuplicateKeyException(
+          duplicateKeyException.getLocalizedMessage(), "Duplicate oid found");
```

Comment:
Do you know if this message includes the duplicate OID value?

Reply:
Yes it does, the localMessage contains oid and description, but I was unable to capture just OID